hack/tools/Makefile

```diff
@@ -66,7 +66,7 @@ $(BIN_DIR)/restart.sh:
 
 GO_APIDIFF := $(BIN_DIR)/go-apidiff
 $(GO_APIDIFF): $(BIN_DIR) go.mod go.sum
-	go build -tags=tools -o $@ github.com/joelanford/go-apidiff
+	go build -tags=tools -o $@ github.com/warehouse-13/go-apidiff
```

Comment:
Could you please revert this change as it's using a fork of github.com/joelanford/go-apidiff repo

Reply:
The main issue is that we don't use `internal` folders and so we get lots of errors with apidiff if we don't try to filter on just the `api` folders.

As @Callisto13 has found out the apidiff tool we use assumes that you use `internal` folders and so a form is created that allows you to filter within the tool itself.

People have tried to filter by grep in the makefile but the existing solution swallows the error.

Reply:
The other option is that we do the work to move everything to an `internal` folder that isn't designed to be consumed outside CAPA. This was discussed at the recent office hours and there was reluctance to this.

Lets start another discussion on this.....

Reply:
Yeah so to use go-apidiff, only best solution left is to move everything to internal, but it would be good to explore more options.
The laziest solution suggested above is one such way to go I guess 🙂 .

Reply:
I am slightly not in favor of forking another repo, as it will be again a maintenance task added to the bucket of maintainers/reviewers.

Reply:
@Ankitasw - i agree on the fork 👍 In the long term i think we need to move things to `internal` but if this other option suggested by @Callisto13  works short term, then great. I wonder what happens if grep returns no matches in this example......

Reply:
> I wonder what happens if grep returns no matches in this example......

Nothing. We don't want the grep to fail, since it will fail the whole job in CI (if there are no matches it means there are no relevant diffs so the job is a noop but does not fail). Since the grep is in the if, the exit code is swallowed. The if branch will only run if grep finds something.

Reply:
updated to used the (refactored) solution I pasted in above